common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java

```diff
@@ -0,0 +1,156 @@
+    private static List<QuestInfo> newQuests;
+
+    public static void init() {}
+
+    @SubscribeEvent
+    public static void onQuestsReloaded(QuestBookReloadedEvent event) {
+        McUtils.player().sendMessage(new TextComponent("Quest Book scanned.").withStyle(ChatFormatting.GRAY), null);
+    }
+
+    @SubscribeEvent
+    public static void onWorldChange(WorldStateEvent e) {
+        if (e.getNewState() == WorldStateManager.State.WORLD) {
+            rescanQuestBook();
+        }
+    }
+
+    /**
+     * Trigger a rescan of the quest book. When the rescan is done, a QuestBookReloadedEvent will
+     * be sent. The available quests are then available using getQuests.
+     */
+    private static void queryQuestBook() {
+        ScriptedContainerQuery.QueryBuilder queryBuilder = ScriptedContainerQuery.builder("Quest Book Query")
+                .onError(msg -> WynntilsMod.warn("Problem querying Quest Book: " + msg))
+                .useItemInHotbar(InventoryUtils.QUEST_BOOK_SLOT_NUM)
+                .matchTitle(getQuestBookTitle(1))
+                .processContainer(c -> processQuestBookPage(c, 1));
+
+        for (int i = 2; i < 5; i++) {
+            final int page = i; // Lambdas need final variables
+            queryBuilder
+                    .clickOnSlotWithName(NEXT_PAGE_SLOT, Items.GOLDEN_SHOVEL, getNextPageButtonName(page))
+                    .matchTitle(getQuestBookTitle(page))
+                    .processContainer(c -> processQuestBookPage(c, page));
+        }
+
+        queryBuilder.build().executeQuery();
+    }
+
+    private static void processQuestBookPage(ContainerContent container, int page) {
+        // Quests are in the top-left container area
+        if (page == 1) {
+            // Build new set of quests without disturbing current set
+            newQuests = new ArrayList<>();
+        }
+        for (int row = 0; row < 6; row++) {
+            for (int col = 0; col < 7; col++) {
+                int slot = row * 9 + col;
+
+                // Very first slot is chat history
+                if (slot == 0) continue;
+
+                ItemStack item = container.items().get(slot);
+                QuestInfo questInfo = QuestInfo.parseItem(item, page);
+                if (questInfo == null) continue;
+
+                newQuests.add(questInfo);
+            }
+        }
+
+        if (page == 4) {
+            // Last page finished
+            quests = newQuests;
+            WynntilsMod.getEventBus().post(new QuestBookReloadedEvent());
+        }
+    }
+
+    public static void trackQuest(QuestInfo questInfo) {
+        ScriptedContainerQuery.QueryBuilder queryBuilder = ScriptedContainerQuery.builder("Quest Book Quest Pin Query")
+                .onError(msg -> WynntilsMod.warn("Problem pinning quest in Quest Book: " + msg))
+                .useItemInHotbar(InventoryUtils.QUEST_BOOK_SLOT_NUM)
+                .matchTitle(getQuestBookTitle(1));
+
+        if (questInfo.getPageNumber() > 1) {
```

Comment:
This case, could you help out with constructing a query which does not need the page number saved? Since this is the second query, perhaps you could do it in a follow up PR (and merge this "buggy")? What do you think?

Reply:
We can certainly start with this query and I can try to improve it later.